src/Phinx/Db/Adapter/SQLiteAdapter.php

```diff
@@ -846,7 +851,15 @@ protected function copyAndDropTmpTable(AlterInstructions $instructions, string $
                 )
             );
 
+            $foreignKeysEnabled = (bool)$this->fetchRow('PRAGMA foreign_keys')['foreign_keys'];
+            if ($foreignKeysEnabled) {
+                $this->execute('PRAGMA foreign_keys = OFF');
+            }
             $this->execute(sprintf('DROP TABLE %s', $this->quoteTableName($tableName)));
+            if ($foreignKeysEnabled) {
+                $this->execute('PRAGMA foreign_keys = ON');
```

Comment:
On a separate note, do we want to enable the foreign key constraint here, or could we move it down to where we validate the FKs, doing:

```php
            if ($foreignKeysEnabled) {
                $this->execute('PRAGMA foreign_keys = ON');
                if ($validateForeignKeys) {
                    $this->validateForeignKeys($tableName);
                }
            }
```

Reply:
I would probably keep it here for now. For #2134 there will be some changes needed to be done, including moving the foreign key check, it will look like this:

https://github.com/ndm2/phinx/blob/0.x-migrations-cake5-test/src/Phinx/Db/Adapter/SQLiteAdapter.php#L1181

I guess the switching could possibly be done in `endAlterByCopyTable()` then too. Maybe we should revisit that in the other PR once it has been rebased/changed accordingly?

Reply:
Sure, that's fine.